gammapy/modeling/models/temporal.py

```diff
@@ -418,5 +470,112 @@ def from_dict(cls, data):
         return cls.read(data["filename"])
 
     def to_dict(self, full_output=False):
-        """Create dict for YAML serilisation"""
+        """Create dict for YAML serialisation"""
         return {"type": self.tag[0], "filename": self.filename}
+
+
+class PowerLawTemporalModel(TemporalModel):
+    r"""Temporal model with a Power Law decay.
+
+    .. math::
+            F(t) = ((t - t_ref)/t0)^alpha
+
+    Parameters
+    ----------
+    alpha : float
+        Decay time power
+    t_ref: `~astropy.units.Quantity`
+        The reference time in mjd. Frozen by default, at 2000-01-01.
+    t0: `~astropy.units.Quantity`
+        The scaling time in mjd. Fixed by default, at 1 day.
+    """
+
+    tag = ["PowerLawTemporalModel", "powerlaw"]
+
+    alpha = Parameter("alpha", 1., frozen=False)
+    _t_ref_default = Time("2000-01-01")
+    t_ref = Parameter("t_ref", _t_ref_default.mjd, unit="day", frozen=True)
+    t0 = Parameter("t0", "1 d", frozen=True)
+
+    @staticmethod
+    def evaluate(time, alpha, t_ref, t0=1*u.day):
+        """Evaluate at given times"""
+        return np.power((time - t_ref)/t0, alpha)
+
+    def integral(self, t_min, t_max):
+        """Evaluate the integrated flux within the given time intervals
+
+        Parameters
+        ----------
+        t_min: `~astropy.time.Time`
+            Start times of observation
+        t_max: `~astropy.time.Time`
+            Stop times of observation
+
+        Returns
+        -------
+        norm : float
+            Integrated flux norm on the given time intervals
+        """
+        pars = self.parameters
+        alpha = pars["alpha"].quantity
+        t0 = pars["t0"].quantity
+        t_ref = Time(pars["t_ref"].quantity, format="mjd")
+        if alpha != -1:
+            value = self.evaluate(t_max, alpha+1., t_ref, t0) - self.evaluate(t_min, alpha+1., t_ref, t0)
+            return t0 / (alpha+1.) * value / self.time_sum(t_min, t_max)
+        else:
+            value = np.log((t_max-t_ref)/(t_min-t_ref))
+            return t0 * value / self.time_sum(t_min, t_max)
+
+
+class SineTemporalModel(TemporalModel):
+    r"""Temporal model with a sinusoidal modulation.
+
+    .. math::
+            F(t) = 1 + amp * sin(omega*(t-t_ref))
+
```

Comment:
Add forward reference as in the other examples...

Reply:
added for the 3